viz-lib/src/visualizations/chart/Editor/SeriesSettings.jsx

```diff
@@ -13,6 +13,9 @@ import getChartData from "../getChartData";
 const SortableBodyRow = sortableElement(props => <tr {...props} />);
 
 function getTableColumns(options, updateSeriesOption, debouncedUpdateSeriesOption) {
+  let restrictedChartTypes = ["pie", "heatmap", "bubble", "box"];
+  if (options.swappedAxes) restrictedChartTypes.push("area");
```

Comment:
_issue_ does not trigger state change

Reply:
@kravets-levko I just noticed an issue from it: the `useMemo` won't recalculate when `options` changes (I think we are just mutating the updates? 🤔), in this case for example the list of available Charts will remain the same in the Series tab, regardless of the Swapping change:

![tabs-wont-update](https://user-images.githubusercontent.com/3356951/96047192-1ace5d00-0e4b-11eb-819d-b81f1bc59422.gif)

---

@rafawendel I'd also not limit the "Area" type from the feature, the result can still be achieved and I don't think it's that important.

Considering that, I recommend you to save yourself from the work of dealing with the possible cases:
<img width="1791" alt="Screen Shot 2020-10-14 at 18 29 37" src="https://user-images.githubusercontent.com/3356951/96047261-39345880-0e4b-11eb-935c-72f1b80aaea6.png">

Also, It feels better not to mess around with previous definitions when one is just testing out the checkbox behavior.

Reply:
I thought so, that's why I preferred to wait your review before proceeding. Also not removing area sounds best, until I can fix the baseline issue some day.

Thank you! I'll push the minor changes ASAP so that we can move on! 😁